ci/ciimage/ubuntu-rolling/install.sh

```diff
@@ -58,6 +59,29 @@ source "$HOME/.cargo/env"
 rustup target add x86_64-pc-windows-gnu
 rustup target add arm-unknown-linux-gnueabihf
 
+# Zig
+# Use the GitHub API to get the latest release information
+LATEST_RELEASE=$(wget -qO- "https://api.github.com/repos/ziglang/zig/releases/latest")
```

Comment:
This is entirely the wrong way to find the latest release and is broken.

Reply:
Not only isn't it broken, your reimplementation using the ziglang.org json file is broken.

I grow weary of nixos-brained people messing things up because they don't understand the basics of design.

Reply:
> Not only isn't it broken, your reimplementation using the ziglang.org json file is broken.

CI failure is coming from the image builder failing which is happening upstream. I'm running a local test to ensure this works locally, from terminal testing it appears to work.

> I grow weary of nixos-brained people messing things up because they don't understand the basics of design.

Um ouch

Reply:
Every distro seems to be cursed with a specific type of user that collects around it.

For Gentoo, of course, it is the people who insist on recompiling all software with [-funroll-loops](https://www.shlomifish.org/humour/by-others/funroll-loops/Gentoo-is-Rice.html) [-fomg-optimize -Ofast](https://wiki.gentoo.org/wiki/GCC_optimization#But_I_get_better_performance_with_-funroll-loops_-fomg-optimize.21).

For Arch, it is "btw I use arch" ricers who who only installed Arch in order to screenshot neofetch and are entirely unfamiliar with running any other aspect of Arch.

And for NixOS it is people who confidently report things are broken, cannot look at or produce logs, and wait for someone else to come point out that it isn't broken and something else is coincidentally broken at the same time but far away -- then ignore that and throw stuff at the wall until something sticks.

And frankly, I think it's pretty lame to tell someone their PR is broken because you "misread" it and didn't even try it, when the alternative changes you propose are actually detrimental to the design  planning. In particular, your alternative inherently defeats the purpose of a CI for "project A" (meson) by instead running CI for "project B" (zig). Don't insult other contributors because you're sad that they didn't download random bizarroland nightly editions of a *compiler* instead of a stable release -- they had their reasoning for choosing stable releases and it's the same reason I share.

I'm sorry but this is just incredibly disappointing, and it is ALWAYS nixos users doing this stuff.

Whatever. I updated the PR myself to exclude setting os_comp.yml, in accordance with my original request that zig be installed in the CI *beforehand*. It also credits the original author.

Since you cannot update the CI image containers at the same time you start depending on them. Our CI is not currently architected in such a way as to allow updating the ci/ciimage/ definitions and have the os_comp.yml jobs use that in the same push -- you have to deliver the updated images to Docker Hub before CI will start running against them.

Reply:
Dude what? I was trying to offer constructive feedback. So what if I misread something, just point that out and leave it be. I didn't insult anyone. Also, don't group every single NixOS user together. I didn't bring up NixOS being anything related to this subject. Your comment feels really uncalled for and if this is how you see contributors, based on what distro they use or what projects they contribute to then I don't feel like it's worth contributing to this project. Meson has always been my favorite build tool as CMake is annoying to use and GNU Make/autotools is even more difficult. I'm just looking to make things better by contributing.

Reply:
That was… unnecessarily mean?

Reply:
> Well, you didn't say what is wrong with it.

That is true, I did make that mistake. I thought this would've 404'd but I misread it.

> But ultimately, I have to say that it goes back to my comment above about "It also credits the original author."

The changes I made are a lot simpler and different but both implementations are only a few lines. That PR also references this one. I was originally going to cherry-pick it but it was just easier to start that commit from scratch. My implementation also works on aarch64 and x86_64 so that's a plus.

> Please take this stuff seriously.

I am, it really feels berating getting this kind of feedback. I already owned up to the mistake I made by editing the message and clarifying on Matrix. Getting this kind of feedback without asking "why is it broken" or how my reimplementation is broken as well doesn't seem constructive. All I've seen is walls of text with little to no substantial feedback.